src/core/positioning/nmeagnssreceiver.cpp

```diff
@@ -0,0 +1,57 @@
+/***************************************************************************
+ nmeagnssreceiver.cpp - NmeaGnssReceiver
+
+ ---------------------
+ begin                : 21.10.2022
+ copyright            : (C) 202 by Matthias Kuhn
+ email                : matthias (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "nmeagnssreceiver.h"
+#include "positioning.h"
+
+#include <QSettings>
+
+NmeaGnssReceiver::NmeaGnssReceiver( QObject *parent )
+  : AbstractGnssReceiver( parent )
+{
+}
+
+void NmeaGnssReceiver::initNmeaConnection( QIODevice *ioDevice )
+{
+  mNmeaConnection = std::make_unique<QgsNmeaConnection>( ioDevice );
+
+  //QgsGpsConnection state changed (received location string)
+  connect( mNmeaConnection.get(), &QgsGpsConnection::stateChanged, this, &NmeaGnssReceiver::stateChanged );
+}
+
+void NmeaGnssReceiver::stateChanged( const QgsGpsInformation &info )
+{
+  if ( mLastGnssPositionValid && std::isnan( info.latitude )               // we already sent a valid position
+       || info.utcDateTime == mLastGnssPositionInformation.utcDateTime() ) // we group updates by timestamp, if the last block is not finished, return
```

Comment:
Is this outdated already?

Reply:
Yes, please review the TCP/UDP receiver PR for thess parts, once merged I'll rebase this PR which will then only contain tests stuff.

Reply:
Done.